src/App.tsx

```diff
@@ -15,33 +15,123 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  let visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPABET) {
+    visibleGoods = visibleGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods = visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+// type State = {
+//   isStarted: boolean;
+//   isReversed: boolean;
+//   sortType: SortType;
+// };
+
+export const App: React.FC = () => {
+  const [isStarted, setIsStarted] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const reverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const sortAlphaBetically = () => {
+    setSortType(SortType.ALPABET);
+  };
+
+  const reset = () => {
+    setIsStarted(false);
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const visible = () => {
+    setIsStarted(!isStarted);
+  };
+
+  return (
+    <div className="App">
+      {!isStarted && (
+        <button
+          type="button"
+          onClick={() => visible()}
+          className="button button__start"
+        >
+          Start
+        </button>
+      )}
+      {isStarted && (
+        <>
+          <ul className="Goods">
+            {getReorderedGoods(goodsFromServer, sortType, isReversed).map(
+              (good) => {
+                return <li key={good}>{good}</li>;
+              },
+            )}
+          </ul>
+
+          <div className="Container">
+            <button
+              type="button"
+              onClick={() => sortAlphaBetically()}
```

Comment:
```suggestion
              onClick={sortAlphaBetically}
```
this approaches are equal